Libplanet/Net/Swarm.cs

```diff
@@ -441,46 +329,104 @@ private set
                 ).Token;
             _cancellationToken = workerCancellationToken;
-            var tasks = new List<Task>();
-            var behindNAT = false;
+            _behindNAT = false;
 
             if (!(_turnClient is null))
             {
                 _publicIPAddress = (await _turnClient.GetMappedAddressAsync()).Address;
 
                 if (await _turnClient.IsBehindNAT())
                 {
-                    behindNAT = true;
+                    _behindNAT = true;
                 }
             }
 
-            if (behindNAT)
+            if (_behindNAT)
             {
                 IPEndPoint turnEp = await _turnClient.AllocateRequestAsync(
                     TurnAllocationLifetime
                 );
                 EndPoint = new DnsEndPoint(turnEp.Address.ToString(), turnEp.Port);
-
-                tasks.Add(BindingProxies(_cancellationToken));
-                tasks.Add(RefreshAllocate(_cancellationToken));
-                tasks.Add(RefreshPermissions(_cancellationToken));
             }
             else
             {
                 EndPoint = new DnsEndPoint(_host, _listenPort.Value);
             }
 
-            using (await _runningMutex.LockAsync())
+            _protocol = new KademliaProtocol<T>(
+                this,
+                _privateKey.PublicKey.ToAddress(),
+                _appProtocolVersion,
+                _cancellationToken,
+                _logger);
+        }
+
+        public async Task StartAsync(
+            int millisecondsBroadcastTxInterval = 5000)
+        {
+            await StartAsync(
+                TimeSpan.FromMilliseconds(millisecondsBroadcastTxInterval)
+            );
+        }
+
+        /// <summary>
+        /// Starts to periodically synchronize the <see cref="BlockChain"/>.
+        /// </summary>
+        /// <param name="broadcastTxInterval">The time period of exchange of staged transactions.
+        /// </param>
+        /// <returns>An awaitable task without value.</returns>
+        /// <exception cref="SwarmException">Thrown when this <see cref="Swarm{T}"/> instance is
+        /// already <see cref="Running"/>.</exception>
+        /// <remarks>If the <see cref="BlockChain"/> has no blocks at all or there are long behind
+        /// blocks to caught in the network this method could lead to unexpected behaviors, because
+        /// this tries to <see cref="IAction.Render"/> <em>all</em> actions in the behind blocks
+        /// so that there are a lot of calls to <see cref="IAction.Render"/> method in a short
+        /// period of time.  This can lead a game startup slow.  If you want to omit rendering of
+        /// these actions in the behind blocks use <see cref=
+        /// "PreloadAsync(IProgress{PreloadState}, IImmutableSet{Address}, CancellationToken)"
+        /// /> method too.</remarks>
+        public async Task StartAsync(TimeSpan broadcastTxInterval)
```

Comment:
`CancellationToken` is missed. is it intended?

Reply:
Cancellation is moved to PrepareAsync, which must be called before StartAsync.

Reply:
`.PrepareAsync()` doesn't seem to use `cancellationToken` directly and it just assigns `_workerCacncellationToken` for later. 

so I think taking `cancellationToken` on `.StartAsync()` and `BootStrapAsync()`  is more natural than as-is.